### ISSUE TITLE
pyspedas.mms_feeps_pad() crashes due to FGM flags

### ISSUE THREAD

**Issue by mshumko** [spedas/pyspedas#593]
# Bug

I can't calculate the FEEPS pitch angle distributions because `pyspedas.mms_feeps_pad()` returns `None` instead of the pytplot variable name.

# Exception(s)
I struggled to track down the bug, as a function call inside `pyspedas.mms_feeps_pad()` was returning `None`, which ultimately propagated to `pyspedas.mms_feeps_pad()` returning `None` as well. 

By stepping through the functions with a debugger, I found the culprit call stack: `mms_feeps_pad()` -> `mms_feeps_pitch_angles()`  -> `mms_load_fgm()`. By default, the `mms_load_fgm` has the `keep_flagged=False` kwarg set, which results in an `IndexError` exception in  `mms_feeps_pitch_angles()` due to the array size mismatch. By overriding the default, `mms_load_fgm(..., keep_flagged=True)` on [L59 of mms_feeps_pitch_angles](https://github.com/spedas/pyspedas/blob/c30b36bb34260a3740853285e8045779dac3fc79/pyspedas/mms/feeps/mms_feeps_pitch_angles.py#L59), the code runs and produces credible looking PADs.

# Discussion

Is the original behavior intended? I feel vary about ignoring flagged data, but I've encountered many time periods that result in this exception. If this is not intended, I can make the change and submit a PR.

# Example to reproduce bug
```python
import pyspedas
import pytplot
import matplotlib.pyplot as plt

time_range = ('2020-06-02T05:00:00', '2020-06-02T06:00:00')
mms_id = 4
energy = (100, 300)

feeds_pad_output = pyspedas.mms_feeps_pad(probe=mms_id, bin_size=30, energy=energy)
feeps_pad = pytplot.data_quants[
    f'mms{mms_id}_epd_feeps_srvy_l2_electron_intensity_{energy[0]}-{energy[1]}keV_pad'
    ]

fig, ax = plt.subplots()
ax.pcolormesh(
    feeps_pad.time, feeps_pad.spec_bins, feeps_pad.values.T, shading='auto'
    )
ax.set_ylabel(f'Pitch angle [deg]')
_text = ax.text(0, 0.99, f'{energy[0]} - {energy[1]} keV', va='top', transform=ax.transAxes)
_text.set_bbox(dict(facecolor='white', linewidth=0, pad=0.1))
plt.show()
```

# Environment
- Python 3.11.4 | packaged by Anaconda, Inc. | (main, Jul  5 2023, 13:47:18) [MSC v.1916 64 bit (AMD64)] on win32
- pyspedas version: 1.4.44

**Comment by jameswilburlewis** [comment]
Thanks for the report!  Is there a missing call to mms_get_feeps in your example?

I pasted it into an interactive Python window and get:
29-Sep-23 16:45:46: The name mms4_epd_feeps_srvy_l2_electron_pitch_angle is currently not in pytplot

Following one of the test cases in tests/feeps.py, I tried this:
    def test_feeps_pad_flagged(self):
        from pytplot import data_quants, tplot_names
        import matplotlib.pyplot as plt
        time_range = ('2020-06-02T05:00:00', '2020-06-02T06:00:00')
        mms_id = 4
        energy = (100, 300)

        mms_load_feeps(probe=mms_id,datatype='electron', trange=time_range, data_rate='srvy')
        feeds_pad_output = mms_feeps_pad(probe=mms_id, bin_size=30, energy=energy)
        tplot_names()
        self.assertTrue(data_exists('mms4_epd_feeps_srvy_l2_electron_intensity_100-300keV_pad'))
        feeps_pad = data_quants[f'mms{mms_id}_epd_feeps_srvy_l2_electron_intensity_{energy[0]}-{energy[1]}keV_pad']
        fig, ax = plt.subplots()
        ax.pcolormesh(
            feeps_pad.time, feeps_pad.spec_bins, feeps_pad.values.T, shading='auto'
            )
        ax.set_ylabel(f'Pitch angle [deg]')
        _text = ax.text(0, 0.99, f'{energy[0]} - {energy[1]} keV', va='top', transform=ax.transAxes)
        _text.set_bbox(dict(facecolor='white', linewidth=0, pad=0.1))
        plt.show()

 
So I've added a call to mms_load_feeps() which I hope is correct for this test.   This seems to work for me, even without the modification to mms_feeps_pitch_angles.py -- the plots I get with keep_flagged=True and keep_flagged=False look identical.

I wonder if it's a Python 3.11 issue, or some other difference in our environments?   I'm using Python 3.9 on an M2 mac, with the latest Pyspedas  master branch and PyTplot matplotlib-backend branch  from our Github repo.   Your version of pyspedas is pretty recent, and I don't think we've changed anything significant in the MMS code in between.    So I will have to dig in a little more to reproduce the issue and understand whether it needs a deeper fix than setting keep_flagged=True when loading fgm data.

**Comment by mshumko** [comment]
Thanks @jameswilburlewis for helping me debug this! 

You are right: I was calling `pyspedas.mms.feeps()` prior to `pyspedas.mms_feeps_pad()` (in a different part of my code). I followed your suggestion and added a `pyspedas.mms_load_feeps()` before the `pyspedas.mms_feeps_pad()` call. This did not fix my problem, however, since the original `pyspedas.mms.feeps()` call created a `tplot` variable that was still crashing the `pyspedas.mms_feeps_pad()` function. 

I suspected that a conflicting tplot variable namespace was the source of my problem. My rather crude solution was to delete all `tplot` variables prior to the `pyspedas.mms_feeps_pad()` call, which worked. In other words:

```python
pytplot.del_data()
pyspedas.mms_load_feeps(
    probe=self.mms_id,
    datatype='electron', 
    trange=[str(t) for t in self.time_range],
    data_rate='srvy'
    )
pyspedas.mms_feeps_pad(probe=self.mms_id, bin_size=bin_size, energy=energy)
```
